### DISCUSSION TITLE
"Website Settings" collection

### DISCUSSION THREAD

**Discussion by CoreyHUN** [pocketbase/pocketbase#2922] [Q&A]
Hello,

I recently started learning PocketBase/NextJS and was wondering what would be the correct way of creating a collection that holds values for website settings like, registrationAllowed, underMaintenance etc? Currently I have a collection named "settings" and a record inside with those 2 bools. What would be the correct / most secure setup of the collection so it's only accessible from the backend itself? 

Should I create a separate admin account that I use from NodeJS, or should I use some sort of API rules? 

Thanks in advance

**Comment by vtisnado** [comment]
The way I usually handle this is creating a collection of **singleton** type, but I didn't see that option in PocketBase. The only collection types are Base, View and Auth.

For the security, the permissions / API rules should be enough. 

Hopefully the maintainers can add the singleton type in the near future.

**Reply by CoreyHUN** [reply]
Hello, thank you for the answer.

Is there a specific API header rule that I could test the request IP against to see if it's localhost or not? I tried testing headers.remote_ip  and x_forwarded_for against 127.0.0.1 but I assume those headers on local requests are not even set? I was checking the docks but couldn't really find anything that I could use for this

If not I'll just stick to an admin account.

**Reply by ganigeorgiev** [reply]
@CoreyHUN The client remote address (I assume that is what you mean by `remote_ip`) is not part of the headers map. `x_forwarded_for` is set usually by proxies and cannot be trusted (unless you are behind a trusted reverse proxy).

I'm not entirely sure that I understand your use case, but if you are planning to interact with PocketBase only server-side, authenticating as admin would be the easiest way to do it. An example admin-only initialized client could be setup in some sort of middleware (_I'm not much familiar with the latest Next.js features so treat the below just as pseudo-code_):

```js
// pb.js
// this file can be imported from everywhere inside your node actions but
// keep in mind that the exported client can be used only as admin and
// it is not intended to deal with regular app user authentication
// ---
import PocketBase, { isTokenExpired } from "pocketbase"

// note: could be loaded from env variables
const ADMIN_EMAIL = "test@example.com"
const ADMIN_PASS = "1234567890"

const pb = new PocketBase("http://127.0.0.1:8090");

// this "middleware" function ideally should be called before each request to ensure that the 
// `pb.authStore` always has valid admin token loaded
export async function authMiddleware() {
  if (pb.authStore.isValid) {
    // just refresh the token if it is going to expire in the next 30 minutes
    if (isTokenExpired(pb.authStore.token, 1800)) {
      await pb.admins.authRefresh()
    }
  } else {
    // authenticate with email/password
    await pb.admins.authWithPassword(ADMIN_EMAIL, ADMIN_PASS)
  }
}

export default pb;

```

**Reply by CoreyHUN** [reply]
Hello,  thank you for the in-depth answer, I'll stick to admin auth and use that then.
